torch/package/importer.py

```diff
@@ -144,7 +145,38 @@ def load_pickle(self, package: str, resource: str, map_location=None) -> Any:
             Any: the unpickled object.
         """
         pickle_file = self._zipfile_path(package, resource)
-        return _load(self.zip_reader, map_location, self, pickle_file=pickle_file)
+        restore_location = _get_restore_location(map_location)
+        loaded_storages = {}
+
+        def load_tensor(data_type, size, key, location, restore_location):
+            name = f'data/{key}'
+            dtype = data_type(0).dtype
+
+            storage = self.zip_reader.get_storage_from_record(name, size, dtype).storage()
+            loaded_storages[key] = restore_location(storage, location)
+
+        def persistent_load(saved_id):
+            assert isinstance(saved_id, tuple)
+            typename = _maybe_decode_ascii(saved_id[0])
+            data = saved_id[1:]
+
+            assert typename == 'storage', \
+                f"Unknown typename for persistent_load, expected 'storage' but got '{typename}'"
+            data_type, key, location, size = data
+            if key not in loaded_storages:
+                load_tensor(data_type, size, key, _maybe_decode_ascii(location), restore_location)
+            storage = loaded_storages[key]
+            return storage
+
+        # Load the data (which may in turn use `persistent_load` to load tensors)
+        data_file = io.BytesIO(self.zip_reader.get_record(pickle_file))
+        unpickler = self.Unpickler(data_file)
+        unpickler.persistent_load = persistent_load
+        result = unpickler.load()
+
+        torch._utils._validate_loaded_sparse_tensors()
```

Comment:
I know that this was already in `_load` and this is just unsplitting code that will diverge, but this stateful weird function will need to be removed in our efforts to unify the format.

Reply:
(it has a race condition if multiple python threads try to read independent files anyway)

Reply:
will add a todo I guess?